Sources/SwiftBSON/BSONDocument.swift

```diff
@@ -517,3 +517,42 @@ extension BSONDocument: BSONValue {
 extension BSONDocument: CustomStringConvertible {
     public var description: String { self.toExtendedJSONString() }
 }
+
+extension BSONDocument {
+    /// Helper function to check for BSON document equality regardless of the order of key-value pairs
+    internal func equalsIgnoreOrderHelper(a: BSONDocument, b: BSONDocument) -> Bool {
+        let keysA = a.keys.sorted()
+        let keysB = b.keys.sorted()
+
+        guard keysA == keysB else {
+            return false
+        }
+
+        for k in keysA {
```

Comment:
rather than iterating over the `keysA` list, it would be more performant to iterate over `a` itself, since the `BSONDocumentIterator` keeps track of where it is in the document and does not need to walk over it from the beginning, like using the subscript (i.e. `a[x]`) does.

Reply:
Oh nice, I didn't realize that 😅